src/Journal.cpp

```diff
@@ -38,11 +43,89 @@ int sLineNumber = -1;
 
 BoolSetting JournalEnabled{ L"/Journal/Enabled", false };
 
+class JournalLogger final
+{
+public:
+   JournalLogger()
+   {
+      wxFileName logFile(FileNames::DataDir(), wxT("journallog.txt"));
+      mLogFile.Open(logFile.GetFullPath(wxPATH_NATIVE), L"w");
+   }
+   void WriteString(std::string_view str)
+   {
+      mLogFile.Write(str.data(), str.size());
+   }
+
+   void FinalizeMessge()
+   {
+      mLogFile.Write("\n");
+      mLogFile.Flush();
+   }
+
+private:
+   wxFFile mLogFile;
+};
+
+JournalLogger& GetLogger()
+{
+   static JournalLogger logger;
+   return logger;
+}
+
+std::string ToString(const wxString& str)
+{
+   return str.ToStdString();
+}
+
+template<typename T>
+std::string ToString(const T& arg)
+{
+   return std::to_string(arg);
+}
+
+template<typename... Args>
+void Log( std::string_view message, const Args... args )
```

Comment:
Maybe `Args && ... args`

Reply:
Yeah, should be const ref, missed that ;-(

Reply:
I was suggesting perfect forwarding which means `&&` but not `const`

Not (necessarily) rvalue references because `Args` is a template parameter.